Files/App.xaml.cs

```diff
@@ -59,6 +68,30 @@ public App()
             StartAppCenter();
         }
 
+        public static void AddHistory(IStorageHistory history)
+        {
+            if (history != null)
+            {
+                StorageHistory.Add(history);
+                if (StorageHistory.Count > 1)
+                {
+                    StorageHistoryIndex++;
+                }
+            }
+        }
+
+        public static void RemoveHistory(IStorageHistory history)
+        {
+            if (history != null)
+            {
+                StorageHistory.Remove(history);
+
+                StorageHistoryIndex--;
+                int index = ArrayHelpers.FitBounds(StorageHistoryIndex, StorageHistory.Count);
+                StorageHistoryIndex = index;
+            }
+        }
+
```

Comment:
Does this need to be put in app.xaml.cs?

Reply:
Hmmm :thinking:. I had the only idea to put it here. Where would you move it?

Reply:
Would it go in a View Model or is there a better way to do it to follow the MVVM pattern?

Reply:
@d2dyno1 Is this code app-wide or just tied to each tab instance? If it's not the latter, you can place all of this inside a new class file. But please stay away from placing stuff like this inside App.cs 🙂

Reply:
@duke7553 ok, I wrapped the mess into separate class, it's much cleaner now